api/src/test/java/org/openmrs/module/eptsreports/reporting/unit/utils/EptsCalculationUtilsTest.java

```diff
@@ -0,0 +1,378 @@
+package org.openmrs.module.eptsreports.reporting.unit.utils;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.openmrs.Encounter;
+import org.openmrs.Obs;
+import org.openmrs.api.APIException;
+import org.openmrs.api.context.ServiceContext;
+import org.openmrs.calculation.patient.PatientCalculationContext;
+import org.openmrs.calculation.patient.PatientCalculationService;
+import org.openmrs.calculation.patient.PatientCalculationServiceImpl;
+import org.openmrs.calculation.result.CalculationResultMap;
+import org.openmrs.calculation.result.ListResult;
+import org.openmrs.calculation.result.ObsResult;
+import org.openmrs.calculation.result.SimpleResult;
+import org.openmrs.module.eptsreports.reporting.calculation.BooleanResult;
+import org.openmrs.module.eptsreports.reporting.cohort.definition.JembiObsDefinition;
+import org.openmrs.module.eptsreports.reporting.helper.TestsHelper;
+import org.openmrs.module.eptsreports.reporting.utils.EptsCalculationUtils;
+import org.openmrs.module.reporting.data.encounter.definition.EncounterDatetimeDataDefinition;
+import org.openmrs.module.reporting.data.patient.EvaluatedPatientData;
+import org.openmrs.module.reporting.data.patient.definition.PatientDataDefinition;
+import org.openmrs.module.reporting.data.patient.service.PatientDataService;
+import org.openmrs.module.reporting.data.person.EvaluatedPersonData;
+import org.openmrs.module.reporting.data.person.definition.ObsForPersonDataDefinition;
+import org.openmrs.module.reporting.data.person.definition.PersonDataDefinition;
+import org.openmrs.module.reporting.data.person.service.PersonDataService;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.EvaluationException;
+import org.openmrs.test.BaseContextMockTest;
+
+public class EptsCalculationUtilsTest extends BaseContextMockTest {
+
+  @Mock private PatientCalculationService service;
+
+  @Spy private TestsHelper testsHelper;
+
+  @Mock private PatientDataService patientDataService;
+
+  @Mock private PersonDataService personDataService;
+
+  @Mock private ServiceContext serviceContext;
+
+  private PatientCalculationContext calculationContext;
+
+  @SuppressWarnings("deprecation")
+  @Before
+  public void init() {
+    contextMockHelper.setService(PersonDataService.class, personDataService);
```

Comment:
Forgot about this, looks like `ContextMockHelper` is deprecated. We should remove `Context.getService` calls from `EptsCalculationUtils`.

Reply:
sure, i hated using it for the same cause, but it looks like the move is tending to use of `BaseModuleContextSensitiveTest` instead `BaseContextMockTest` whenever testing something that tends to do `Context.*` requests.

Reply:
I don't think it will be necessary to use `BaseModuleContextSensitiveTest`. I can see only 2 calls to `Context.getSevice` in `EptsCalculationUtils` which are already mocked/stubbed in these tests.

I think we only need to define the services as fields and some king of injection like they describe here: 
https://github.com/openmrs/openmrs-core/blob/4a0feb8da351088f25fdc4e6d324a1f277aa3410/api/src/test/java/org/openmrs/api/context/ContextMockHelper.java#L42-L55

Reply:
just switched to run with powermock so as to statically mock Context.